### ISSUE TITLE
EnpkgVersion mangles invalid version numbers to be valid

### ISSUE THREAD

**Issue by sjagoe** [enthought/okonomiyaki#37]
okonomiyaki changes the meaning of version numbers and then pretends it has not been worked around.

I would expect `enpkg_version.upstream.is_worked_around` to be `True` in this example:

```
>>> enpkg_version = EnpkgVersion.from_upstream_and_build(
    '4.5.0.dev', 1)
>>> print(str(enpkg_version.upstream))
4.5.0.dev1
>>> print(enpkg_version.upstream.is_worked_around)
False
```

**Comment by cournape** [comment]
This is done there: https://github.com/enthought/enstaller_private/blob/master/enstaller/versions/pep386_workaround.py

The reason this is not considered worked around is that enpkg has assumed  `...dev` == `...dev1` for a long time, and setting it up as worked around may mean breaking some existing version comparisons. We should double check though

**Comment by sjagoe** [comment]
Then I am going to need to check versions against the internals of this. I don't know if the `EnpkgVersion` even keeps its original input to be able to determine this.

What else depends on `is_worked_around`? Surely keeping the version number mangling but setting `is_worked_around` should be safe?

**Comment by sjagoe** [comment]
The fact that I can't easily detect that my version number is changing underneath me must be a bug. I would suggest changing the is_worked_around handling to the below (that is, mangle/"normalize" the version number only if there is an `IrrationalVersionError`.

```
class PEP386WorkaroundVersion(object):
    """A version class that supports comparison, with an escape for
    versions not compatible with PEP386.
    When comparing two versions, 3 cases arise:
    * both are valid: we use the PEP386 comparison algo
    * both are invalid: we use string comparison
    * exactly one of them is valid: the valid one is always considered
    to be greather than the invalid one
    """
    @classmethod
    def from_string(cls, s):
        try:
            version = NormalizedVersion(
                s, error_on_huge_major_num=False)
            parts = version.parts
            return cls(parts)
        except IrrationalVersionError:
            normalized = normalize_version_string(s)
            parts = normalized.split(".")
            return cls(parts, is_worked_around=True)
```

**Comment by sjagoe** [comment]
Closed by #39
